src/BufferedTokenStream.ts

```diff
@@ -308,10 +308,6 @@ export class BufferedTokenStream implements TokenStream {
 			throw new RangeError("start " + start + " or stop " + stop + " not in 0.." + (this.tokens.length - 1));
 		}
 
-		if (start === 0 && stop === this.tokens.length - 1) {
```

Comment:
📝 So this was supposed to handle the case of this method:

https://github.com/tunnelvisionlabs/antlr4/blob/be017394a160f85d201f55aac962090e30f1b4b6/runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java#L267

It looks like I messed this up a bit in a few ways. Let me add back the explicit overloads and implement this in a manner more consistent with patterns followed later in the porting work.